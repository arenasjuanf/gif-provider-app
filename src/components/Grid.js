import React, { useState, useEffect } from 'react';

export const Grid = ({category}) => {

    const [count, setCount] = useState(0);
    useEffect(() => { // ejecuta la funciÃ³n solo la primera vez que se renderiza
        getGifs();
    },[]);

    const getGifs = async () => {

        const url = 'http://api.giphy.com/v1/gifs/trending?api_key=wxZG2HVTt9mOB6Xhx55UTAi6migipmOF&q=halo&limit=15';
        const resp = await fetch(url);
        const {data} = await resp.json();
        const gifs = data.map(img => {

            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gifs);
    };


    return (
        <div>
            <h3>
                { category }
            </h3>
        </div>
    )
}

export default Grid;