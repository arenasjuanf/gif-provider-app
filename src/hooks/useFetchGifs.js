
import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';


export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => { // ejecuta la funciÃ³n solo la primera vez que se renderiza
        getGifs(category).then((data)  =>setState({data, loading: false}))
    },[category]);



    return state; //{ data: [], loading: true }
}