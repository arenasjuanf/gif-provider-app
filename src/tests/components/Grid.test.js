import { shallow } from 'enzyme';
import Grid from '../../components/Grid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import "@testing-library/jest-dom";
jest.mock('../../hooks/useFetchGifs');

describe("<Grid> Tests", () =>{

    const category = "spiderman";

    test("Grid component should show successfully", () => {

        useFetchGifs.mockReturnValue({
            loading: true,
            data:[]
        });

        const wrapper = shallow(<Grid category={ category }/>);
        expect(wrapper).toMatchSnapshot();

    })

    test("should show items when images are ready", () => {

        const data = [
            {id:"abc", url: "https://localhost/test", title: "title"}
        ];

        useFetchGifs.mockReturnValue({
            loading: false,
            data
        });

        const wrapper = shallow(<Grid category={ category }/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GridGifItem').length).toBe(data.length);

    })

})