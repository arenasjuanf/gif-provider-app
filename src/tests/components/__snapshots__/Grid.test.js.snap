// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid> Tests Grid component should show successfully 1`] = `
<Fragment>
  <h3>
    spiderman
  </h3>
  <p
    className="animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
  <hr />
</Fragment>
`;

exports[`<Grid> Tests should show items when images are ready 1`] = `
<Fragment>
  <h3>
    spiderman
  </h3>
  <div
    className="card-grid"
  >
    <GridGifItem
      id="abc"
      key="abc"
      title="title"
      url="https://localhost/test"
    />
  </div>
  <hr />
</Fragment>
`;
